api/src/Entity/ContentNode/ColumnLayout.php

```diff
@@ -73,6 +75,45 @@ class ColumnLayout extends ContentNode {
     #[ORM\Column(type: 'json', nullable: true)]
     public ?array $columns = [['slot' => '1', 'width' => 12]];
 
+    /**
+     * All content nodes that are part of this content node tree.
+     */
+    #[ApiProperty(readable: false, writable: false)]
+    #[ORM\OneToMany(targetEntity: ContentNode::class, mappedBy: 'root')]
+    public Collection $rootDescendants;
+
+    public function __construct() {
+        parent::__construct();
+        $this->rootDescendants = new ArrayCollection();
+    }
+
+    /**
+     * @return ContentNode[]
+     */
+    public function getRootDescendants(): array {
+        return $this->rootDescendants->getValues();
+    }
+
+    public function addRootDescendant(ContentNode $rootDescendant): self {
+        if (!$this->rootDescendants->contains($rootDescendant)) {
+            $this->rootDescendants[] = $rootDescendant;
+            $rootDescendant->root = $this;
+        }
+
+        return $this;
+    }
+
+    public function removeRootDescendant(ContentNode $rootDescendant): self {
+        if ($this->rootDescendants->removeElement($rootDescendant)) {
+            // reset the owning side (unless already changed)
+            if ($rootDescendant->root === $this) {
+                $rootDescendant->root = null;
+            }
+        }
+
+        return $this;
+    }
```

Comment:
> These will also be available on nested column layouts, and appear in the API (?). We need to take care to only use these methods on root content nodes.

`rootDescendants` will be available on all ColumnLayout entities, but empty for nested ones (non-root entities). Not displaying this for nested ColumnLayouts would break the rule, that an entity of the same type always looks the same.

> Or maybe it would help to create another dedicated root content node type instead of using `ColumnLayout` for it.

We discussed this as an option in https://github.com/ecamp/ecamp3/issues/2569 and, yes, I have the opinion it would be cleaner. In the meeting we settled on using `ColumnLayout` as root node for the first shot.

This would however become irrelevant, in case we decide to implement https://github.com/ecamp/ecamp3/issues/2633